src/main/java/org/eclipse/yasson/internal/serializer/MapDeserializer.java

```diff
@@ -116,34 +119,43 @@ public void appendResult(Object result) {
             ((Map<String, V>) getInstance(null)).put(key, value);
         }
     }
-    
+
     @SuppressWarnings("unchecked")
     private <K> K asStringable(Class<K> clazz, String value) {
-        Method found = null;
-        try {
-            Method m = clazz.getMethod("valueOf", String.class);
-            if (Modifier.isStatic(m.getModifiers()) && Modifier.isPublic(m.getModifiers()))
-                found = m;
-        } catch (NoSuchMethodException ignore) {
-        }
-        if (found == null) {
-        try {
-            Method m = clazz.getMethod("fromString", String.class);
-            if (Modifier.isStatic(m.getModifiers()) && Modifier.isPublic(m.getModifiers()))
-                found = m;
-        } catch (NoSuchMethodException ignore) {
-        }
-        }
-        if (found != null) {
+        Optional<Method> method = stringableMethodCache.computeIfAbsent(clazz, c -> {
+            Method valueOf = getStaticPublicMethod(c, "valueOf");
+            if (valueOf != null) {
+                return Optional.of(valueOf);
+            }
+            Method fromString = getStaticPublicMethod(c, "fromString");
+            if (fromString != null) {
+                return Optional.of(fromString);
+            }
+            return Optional.empty();
+        });
+
+        if (method.isPresent()) {
             try {
-                return (K) found.invoke(null, value);
+                return (K) method.get().invoke(null, value);
             } catch (IllegalAccessException | IllegalArgumentException | InvocationTargetException e) {
                 throw new JsonbException(Messages.getMessage(MessageKeys.DESERIALIZE_VALUE_ERROR, value, clazz), e);
             }
         }
         return null;
     }
-    
+
+    private Method getStaticPublicMethod(Class<?> clazz, String name) {
+        try {
+            Method method = clazz.getMethod(name, String.class);
+            if (Modifier.isStatic(method.getModifiers()) && Modifier.isPublic(method.getModifiers())) {
+                method.setAccessible(true);
```

Comment:
We may not want to use setAccessible. Although we know the method is public I think this can still throw SecurityException if the class is not public